src/main/webapp/app/entities/modeling-assessment/modeling-assessment.service.ts

```diff
@@ -71,30 +76,21 @@ export class ModelingAssessmentService {
         return copy;
     }
 
-    private convertAssessmentFromServer(assessment: ModelingAssessment): ModelingAssessment {
-        const copy: ModelingAssessment = Object.assign({}, assessment);
+    private convertAssessmentFromServer(result: Result): Result {
+        const copy: Result = Object.assign({}, result);
         return copy;
     }
 
-    private convertArrayResponse(res: HttpResponse<ModelingAssessment[]>): HttpResponse<ModelingAssessment[]> {
-        const jsonResponse: ModelingAssessment[] = res.body;
-        const body: ModelingAssessment[] = [];
-        if (jsonResponse) {
-            for (let i = 0; i < jsonResponse.length; i++) {
-                body.push(this.convertAssessmentFromServer(jsonResponse[i]));
-            }
-        }
-        return res.clone({body});
-    }
-
     /**
      * Creates the labels for the assessment elements for displaying them in the modeling and assessment editor.
      */
-    getNamesForAssessments(assessments: ModelingAssessment[], model: State): Map<string, Map<string, string>> {
+    getNamesForAssessments(result: Result, model: State): Map<string, Map<string, string>> {
         const assessmentsNames = new Map<string, Map<string, string>>();
-        for (const assessment of assessments) {
-            if (assessment.type === ModelElementType.CLASS) {
-                const classElement = model.entities.byId[assessment.id];
+        for (const feedback of result.feedbacks) {
+            const referencedModelType = feedback.referenceType;
+            const referencedModelId = feedback.referenceId;
+            if (feedback.reference === ModelElementType.CLASS) {
```

Comment:
I think the referencedModelType should be checked here instead of the full reference: `if (referencedModelType === ModelElementType.CLASS) {` 
Or am I missing something?

Reply:
yes, please fix it directly

Reply:
done